packages/chain-helpers/src/blockchain/Blockchain.ts

```diff
@@ -174,19 +176,47 @@ export const dispatchTx = submitSignedTx
  * @param signer The {@link KeyringPair} used to sign the tx.
  * @param opts Additional options.
  * @param opts.tip Optional amount of Femto-KILT to tip the validator.
+ * @param opts.checkMetadata Boolean flag indicated whether to verify the metadata hash upon tx submission.
  * @returns A signed {@link SubmittableExtrinsic}.
  */
 export async function signTx(
   tx: SubmittableExtrinsic,
   signer: KeyringPair | TransactionSigner,
-  { tip }: { tip?: AnyNumber } = {}
+  { tip, checkMetadata }: { tip?: AnyNumber; checkMetadata?: boolean } = {}
 ): Promise<SubmittableExtrinsic> {
+  const signOptions = await (async (): Promise<Partial<SignerOptions>> => {
+    if (!checkMetadata) {
+      return {
+        tip,
+      }
+    }
+    // If `checkMetadata` is enabled, include that when signing the tx.
+    const api = ConfigService.get('api')
+    const metadata = await api.call.metadata.metadataAtVersion(15)
```

Comment:
the api should have the metadata cached, can't we use this one?

Reply:
Good point: https://github.com/KILTprotocol/sdk-js/pull/924/commits/7b9e1e4294504ba65287bf6db174849069d3a998.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/924#discussion_r1913013747